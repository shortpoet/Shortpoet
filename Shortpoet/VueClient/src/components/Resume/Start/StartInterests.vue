<template>
  <section class="resume-section p-3 p-lg-5 d-flex flex-column" id="interests">
    <div class="my-auto">
      <h2 class="mb-4 mb-md-3">Interests</h2>
      <p class="mb-4">{{ interests }}</p>
    </div>
    <div class="short-poetry my-auto">
      <!-- <div class='embed-container'><iframe src='//instagram.com/p/BK2nW22Bg1l/embed/' frameborder='0' scrolling='no' allowtransparency='true'></iframe></div>     -->
      <div id="poem-container" class="mt-3 mt-md-0 mr-0 text-center">
        <img class="img-fluid img-profile rounded" :src="pic" alt="poem">
      </div>
    </div>
  </section>
</template>
<script>
export default {
  name: 'StartInterests',
  props: {
    interests: {
      type: String
    }
  },
  data () {
    return {
      pic: require('@/assets/poem.jpg'),
    }
  }
}
</script>

<style scoped>
#interests {
  align-items: center;
}
.short-poetry {
  height: 50%;
  width: 50%;  
}
.embed-container { position: relative; padding-bottom: 150%; height: 0; overflow: hidden; max-width: 100%; } 
.embed-container iframe, .embed-container object, .embed-container embed { position: absolute; top: 0; left: 0; width: 100%; height: 100%; }
</style>
