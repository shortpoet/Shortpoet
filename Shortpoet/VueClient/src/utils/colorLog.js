const chalk = require('chalk')
import {inspect} from 'util'
export const log = (color, message) => console.log(chalk[`${color}`](`${inspect(message)}`))

export const colorLog = (message, color, background) => {
  color = color || "black";
  if (background) {
    console.log(`%c` + `${message}`, `color:` + `${color};background:${background}`)
  } else {
    console.log(`%c` + `${message}`, `color:` + `${color}`)
  }
}

// module.exports =  {
//   log,
//   colorLog
// }
