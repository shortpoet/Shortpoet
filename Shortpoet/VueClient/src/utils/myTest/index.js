// address
import createWrapper from './test.utils.createWrapper'
import factory from './test.utils.factory'
import factoryShallow from './test.utils.factoryShallow'
export default {
  createWrapper,
  factory,
  factoryShallow
}
