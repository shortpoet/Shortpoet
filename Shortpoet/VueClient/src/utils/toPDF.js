import jsPDF from 'jspdf'
// using fork for now to solve this issue
// the changes i made with the icons as the deployed version still works
// https://github.com/niklasvh/html2canvas/issues/1868#issuecomment-599217709
import html2canvas from '@trainiac/html2canvas'

const toPDF = (target) => {
  // timeout is set to account for loading time i believe
  setTimeout(() => {
    console.log(target)
    html2canvas(document.getElementById(target), {
      scale: 3,
      useCORS: true,
      allowTaint: true,
    }).then(canvas => {
      const image = canvas.toDataURL('image/jpeg', 1.0);
      const doc = new jsPDF('p', 'mm', 'a4');
      const pageWidth = doc.internal.pageSize.getWidth();
      const pageHeight = doc.internal.pageSize.getHeight();

      const widthRatio = pageWidth / canvas.width;
      const heightRatio = pageHeight / canvas.height;
      const ratio = widthRatio > heightRatio ? heightRatio : widthRatio;

      const canvasWidth = canvas.width * ratio;
      const canvasHeight = canvas.height * ratio;

      const marginX = (pageWidth - canvasWidth) / 2;
      const marginY = (pageHeight - canvasHeight) / 2;

      doc.addImage(image, 'JPEG', marginX, marginY, canvasWidth, canvasHeight, null, 'SLOW');
      doc.save(`Carlos_Soriano_${Date.now()}.pdf`);
    });
  }, 250);
}

export default toPDF