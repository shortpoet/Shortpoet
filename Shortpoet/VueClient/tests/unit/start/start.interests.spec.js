import { textMatcher, propsMocker, createWrapper } from '../test.utils'
import StartInterests from '@/components/Resume/Start/StartInterests'

describe('StartInterests.vue', () => {

  const component = StartInterests
  const props = Object.keys(component.props)
  const wrapper = createWrapper(component, propsMocker(props))

  let prop
  let selector
  let mockProp = true

  beforeEach(() => {
    jest.resetModules()
    jest.clearAllMocks()
  })

  it('renders interests h2', () => {

    // hard coded header
    expect(wrapper.find("h2").text()).toMatch("Interests")

  })
  it('renders interests p that matches interests prop', () => {

    prop = 'interests'
    selector = 'p'
    textMatcher(component, props, prop, selector)

  })
  it('renders photo', () => {

    expect(wrapper.find("img").exists()).toBe(true)

  })
  it('matches snapshot', () => {
    
    expect(wrapper.html()).toMatchSnapshot()

  })
})
