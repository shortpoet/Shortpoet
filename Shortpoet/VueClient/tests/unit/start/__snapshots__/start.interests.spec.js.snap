// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartInterests.vue matches snapshot 1`] = `
<section id="interests" class="resume-section p-3 p-lg-5 d-flex flex-column">
  <div class="my-auto">
    <h2 class="mb-4 mb-md-3">Interests</h2>
    <p class="mb-4">sample interests</p>
  </div>
  <div class="short-poetry my-auto">
    <div id="poem-container" class="mt-3 mt-md-0 mr-0 text-center"><img src="" alt="poem" class="img-fluid img-profile rounded"></div>
  </div>
</section>
`;
